modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/CommandProcessor.java

```diff
@@ -725,15 +725,18 @@ private static QueryEntity toQueryEntity(GridSqlCreateTable createTbl) {
                 dfltValues.put(e.getKey(), dfltVal);
 
             if (col.getType() == Value.DECIMAL) {
-                precision.put(e.getKey(), (int)col.getPrecision());
+                if (col.getPrecision() != H2Utils.DECIMAL_DEFAULT_PRECISION)
```

Comment:
let's use 'less sign' instead of 'not equals'. Such approach use H2, because precision has type long.